tests/e2e/specs/miscellaneous/VsixInstallationDisableTest.spec.ts

```diff
@@ -0,0 +1,283 @@
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+		workspaceName = WorkspaceHandlingTests.getWorkspaceName();
+		registerRunningWorkspace(workspaceName);
+	});
+
+	test('Wait workspace readiness', async function (): Promise<void> {
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+	});
+
+	test('Verify VSIX disabled notifications', async function (): Promise<void> {
+		const hasDisableNotification: boolean = await notificationHandler.checkForNotification('install from vsix command is disabled');
+		expect(hasDisableNotification).to.be.true;
+	});
+
+	test('Perform trust author dialog', async function (): Promise<void> {
+		await projectAndFileTests.performTrustAuthorDialog();
+	});
+
+	test('Verify VSIX installation is disabled in UI', async function (): Promise<void> {
+		await checkVsixInstallationInUI(false);
+	});
+
+	test('Verify default extensions are not auto-installed when VSIX disabled', async function (): Promise<void> {
+		await checkDefaultExtensionsInstallation(false);
+	});
+
+	test('Enable VSIX installation and create new workspace', async function (): Promise<void> {
+		// clean up current workspace
+		await dashboard.openDashboard();
+		await testWorkspaceUtil.stopAndDeleteWorkspaceByName(workspaceName);
+		registerRunningWorkspace('');
+
+		// apply new ConfigMaps
+		applyConfigMap(CONFIG_FILES.ENABLE_VSIX);
+		applyConfigMap(CONFIG_FILES.DEFAULT_EXTENSIONS);
+
+		Logger.info('Waiting for new ConfigMap settings to take effect...');
+		await driverHelper.wait(30000);
+
+		// create new workspace
+		await workspaceHandlingTests.createAndOpenWorkspaceFromGitRepository(TEST_REPO_URL);
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+		workspaceName = WorkspaceHandlingTests.getWorkspaceName();
+		registerRunningWorkspace(workspaceName);
+	});
+
+	test('Wait workspace readiness', async function (): Promise<void> {
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+	});
+
+	test('Verify default extension installation success notifications', async function (): Promise<void> {
+		const successTexts: string[] = ['Completed installing extension', 'installed', 'extension'];
+		const hasSuccessNotification: boolean = await notificationHandler.checkForAnyNotification(successTexts);
+		expect(hasSuccessNotification).to.be.true;
+	});
+
+	test('Perform trust author dialog', async function (): Promise<void> {
+		await projectAndFileTests.performTrustAuthorDialog();
+	});
+
+	test('Verify VSIX installation is enabled in UI', async function (): Promise<void> {
+		await checkVsixInstallationInUI(true);
+	});
+
+	test('Verify default extensions are auto-installed when VSIX enabled', async function (): Promise<void> {
+		await checkDefaultExtensionsInstallation(true);
+	});
+
+	suiteTeardown('Clean up ConfigMaps and workspace', async function (): Promise<void> {
+		cleanupConfigMaps();
```

Comment:
IMHO, it makes sense to repeat the steps to enable VSIX installation in the teardown, in case the test failed in the middle of execution.